File: solutions/c/matching-brackets/1/matching_brackets.c
#include "matching_brackets.h"
#include <stdio.h>

bool is_paired(const char *input) {
    char br[200];
    int bi=-1;
    for (int i=0; input[i]!='\0'; i++) {
        if (input[i] == '['||input[i] == ']'||input[i] == '{'||input[i] == '}'||input[i] == '('||input[i] == ')') br[++bi]=input[i];
        if (bi>0 && ((br[bi-1]=='['&&br[bi]==']')||(br[bi-1]=='{'&&br[bi]=='}')||(br[bi-1]=='('&&br[bi]==')'))) bi-=2;
    }
    if (bi>=0) return false;
    return true;
}

File: solutions/csharp/matching-brackets/1/MatchingBrackets.cs
using System;
using System.Text.RegularExpressions;

public static class MatchingBrackets
{
    public static bool IsPaired(string input)
    {

        Regex nobr = new Regex("[^\\[\\]{}()]+");
        string ret=input;
        var matches = nobr.Matches(ret);
        while (matches.Count > 0)
        {
            ret = Regex.Replace(ret,"[^\\[\\]{}()]+", "");
            matches = nobr.Matches(ret);
        }
        Regex blbr = new Regex("\\(\\)|{}|\\[\\]");
        matches = blbr.Matches(ret);
        while (matches.Count > 0)
        {
            ret = Regex.Replace(ret, "\\(\\)|{}|\\[\\]", "");
            matches = blbr.Matches(ret);
        }
        if (ret.Length>0) return false;
        return true;
    }


}
